src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/SemanticTokensCache.cs

```diff
@@ -115,6 +122,11 @@ internal sealed class SemanticTokensCache
             var tokens = GetCachedTokens(requestedRange, lineToTokensDict, out var cachedRangeStart, out var numLinesInCachedRange);
             if (tokens.Length == 0)
             {
+                if (logger is not null)
+                {
+                    logger.LogInformation($"No cached results found for range: {requestedRange}");
```

Comment:
Instead of doing `logger is not null` could do `logger?.LogInformation`. 

Also, mind be worth changing this log text so you can differentiate this log from the one added above